src/common/meta/src/kv_backend/rds/mysql.rs

```diff
@@ -0,0 +1,648 @@
+        let query = sqlx::query(raw_query);
+        let query = params.iter().fold(query, |query, param| query.bind(param));
+        // As said in https://docs.rs/sqlx/latest/sqlx/trait.Executor.html, we need a `&mut *transaction`. Weird.
+        let rows = query
+            .fetch_all(&mut *(self.0))
+            .await
+            .context(MySqlExecutionSnafu { sql: raw_query })?;
+        Ok(rows.into_iter().map(key_value_from_row).collect())
+    }
+
+    async fn execute(&mut self, raw_query: &str, params: &[&Vec<u8>]) -> Result<()> {
+        let query = sqlx::query(raw_query);
+        let query = params.iter().fold(query, |query, param| query.bind(param));
+        // As said in https://docs.rs/sqlx/latest/sqlx/trait.Executor.html, we need a `&mut *transaction`. Weird.
+        query
+            .execute(&mut *(self.0))
+            .await
+            .context(MySqlExecutionSnafu { sql: raw_query })?;
+        Ok(())
+    }
+
+    /// Caution: sqlx will stuck on the query if two transactions conflict with each other.
+    /// Don't know if it's a feature or it depends on the database. Be careful.
+    async fn commit(self) -> Result<()> {
+        self.0.commit().await.context(MySqlTransactionSnafu {
+            operation: "commit",
+        })?;
+        Ok(())
+    }
+}
+
+pub struct MySqlExecutorFactory {
+    pool: Arc<Pool<MySql>>,
+}
+
+#[async_trait::async_trait]
+impl ExecutorFactory<MySqlClient> for MySqlExecutorFactory {
+    async fn default_executor(&self) -> Result<MySqlClient> {
+        Ok(self.pool.clone())
+    }
+
+    async fn txn_executor<'a>(
+        &self,
+        default_executor: &'a mut MySqlClient,
+    ) -> Result<MySqlTxnClient> {
+        default_executor.txn_executor().await
+    }
+}
+
+/// A MySQL-backed key-value store.
+/// It uses [sqlx::Pool<MySql>] as the connection pool for [RdsStore].
+pub type MySqlStore = RdsStore<MySqlClient, MySqlExecutorFactory, MySqlTemplateSet>;
+
+#[async_trait::async_trait]
+impl KvQueryExecutor<MySqlClient> for MySqlStore {
+    async fn range_with_query_executor(
+        &self,
+        query_executor: &mut ExecutorImpl<'_, MySqlClient>,
+        req: RangeRequest,
+    ) -> Result<RangeResponse> {
+        let template_type = range_template(&req.key, &req.range_end);
+        let template = self.sql_template_set.range_template.get(template_type);
+        let params = template_type.build_params(req.key, req.range_end);
+        let params_ref = params.iter().collect::<Vec<_>>();
+        // Always add 1 to limit to check if there is more data
+        let query =
+            RangeTemplate::with_limit(template, if req.limit == 0 { 0 } else { req.limit + 1 });
+        let limit = req.limit as usize;
+        debug!("query: {:?}, params: {:?}", query, params);
+        let mut kvs = query_executor.query(&query, &params_ref).await?;
+        if req.keys_only {
+            kvs.iter_mut().for_each(|kv| kv.value = vec![]);
+        }
```

Comment:
I think we should make `keys_only` faster by using a different SQL query, such as `select k from ..`, without including v. We can add a `TODO` to address this in the future, if not in this PR.